packages/template-retail-react-app/app/commerce-api/hooks/useCustomer.js

```diff
@@ -10,6 +10,8 @@ import {useCommerceAPI, CustomerContext} from '../contexts'
 
 const AuthTypes = Object.freeze({GUEST: 'guest', REGISTERED: 'registered'})
 
+const REGISTRATION_GRACE_PERIOD = 2 * 1000 // 2 seconds in milliseconds
```

Comment:
"Grace period" is confusing - is this just a workaround for "just registered _and_ logged in"?

Reply:
Is this is a work around for determining if a user has just registered. Unfortunately there isn't a property on the customer object that tells you that directly. There is a host of data type values, but they unfortunately cant be used to derive whether or not a user is new. 

I was struggling with that name, I'm open to suggestions on other names, or solutions.